src/EditorFeatures/TestUtilities/Extensions/IDiagnosticServiceExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.Extensions;
+
+internal static class IDiagnosticServiceExtensions
+{
+    public static Task<ImmutableArray<DiagnosticData>> ForceAnalyzeProjectAsync(
```

Comment:
unit test only extension for tests that want to validate the diags that CodeAnalysisDiagnosticAnalyzerService gets back.  I will do a future pr that has these features literally just test that service instead of diving into one of its helper methosd.